test/common/config/registry_test.cc

```diff
@@ -103,6 +103,19 @@ TEST(RegistryTest, DEPRECATED_FEATURE_TEST(WithDeprecatedFactoryPublished)) {
                           ->name());
 }
 
+class TestWithDeprecatedPublishedFactoryNoDeprecatedNames : public PublishedFactory {
+public:
+  std::string name() const override { return ""; }
+};
+
+TEST(RegistryTest, DEPRECATED_FEATURE_TEST(AssertsIfNoDeprecatedNameGiven)) {
+  // Expects an assert to raise if we register a factory that has an empty name
+  // and no associated deprecated names.
+  EXPECT_DEATH((Registry::RegisterFactory<TestWithDeprecatedPublishedFactoryNoDeprecatedNames,
+                                          PublishedFactory>({})),
+               ".*assert failure:.*");
```

Comment:
Is there a more specific string you can expect on death?  You can add a second argument to the ASSERT macro so you have a string you can expect here.
ASSERT(deprecated_names.size() != 0, "PublishedFactory has no name or deprecated name");

Also, I think that the .* are redundant; EXPECT_DEATH does search for a partial match of the provided failure message.